src/lib.rs

```diff
@@ -319,84 +325,84 @@ impl<'a> ClamdClientBuilder<'a> {
 /// # use eyre::Result;
 /// # async fn doc() -> eyre::Result<()> {
 /// let address = "127.0.0.1:3310";
-/// let mut clamd_client = ClamdClientBuilder::tcp_socket(&address.parse()?).build();
+/// let mut clamd_client = ClamdClientBuilder::tcp_socket(address)?.build();
 /// clamd_client.ping().await?;
 /// # Ok(())
 /// # }
 /// ```
 pub struct ClamdClient {
```

Comment:
Small thing: now that this is threadable, please add `#[derive(Clone)]` (you can always derive clone for struct if all of its members are also cloneable which is always true for an Arc. Also you could just use a tuple struct here if we have only one element, i.e. `ClamdClient(Shared)` but thats not that important.

Reply:
Done, and as for the tuple struct, I personally prefer in this case to have a named member